test/rack/prx_auth_test.rb

```diff
@@ -59,15 +61,47 @@
     end
   end
 
-  describe '#token_expired?' do
-    it 'returns true if token is expired' do
-      claims['iat'] = Time.now.to_i - 4000
+  describe '#expired?' do
 
-      assert prxauth.send(:expired?, claims) == true
+    def expired?(claims)
+      prxauth.send(:expired?, claims)
     end
 
-    it 'returns false if it is valid' do
-      assert prxauth.send(:expired?, claims) == false
+    it 'is expired if the token is issued in the future' do
+      claims['iat'] = Time.now.to_i + 3600
+
+      assert expired?(claims)
+    end
+
+    describe 'with a malformed exp' do
+      let(:exp) { 3600 }
+
+      it 'is expired if iat + exp are in the past' do
+        claims['iat'] -= 3601
+
+        assert expired?(claims)
+      end
+
+      it 'is not expired if iat + exp are in the future' do
+        claims['iat'] -= 3599
```

Comment:
I see - and this is the backwards-compatible case, for tokens that are out there already when this deploys.